### PR TITLE
Rewrite home page as a client component

### PR DESCRIPTION
**Pull request by henrycatalinismith**
How about this?

* No server-side data access outside the REST API.
* No dodgy-looking Next.js 14 upgrade.
* No interesting new Next.js routing features.

Just a good old fashioned bit of client-side rendering!

### CODE REVIEW

File: webapp/src/app/page.tsx [hunk 1]
```diff
@@ -1,57 +1,62 @@
-import { Cache } from '@/Cache';
-import HomeDashboard from '@/components/HomeDashboard';
```
Comment (WULCAN) [inline webapp/src/app/page.tsx:2]: This was the only use of HomeDashboard. Let's remove it now that we have inlined it on this page.

File: webapp/src/app/page.tsx [hunk 2]
```diff
@@ -1,57 +1,62 @@
...
+};
 
-      return {
-        href: `/projects/${project.name}`,
-        languages,
-        messageCount: messages.length,
-        name: project.name,
-      };
-    }),
-  );
```
Comment (WULCAN) [inline webapp/src/app/page.tsx:54]: The old implementation has been moved, almost verbatim, to `api/projects/route.ts` with the only difference being that `ProjectCardProps` is replaced by `ProjectItem`. Those two types are exactly equal.

**Review by WULCAN — CHANGES_REQUESTED** [review]
Thank you, this is excellent work. I just want us to clean up the killed code too. See the thread on `src/app/page.tsx`.

We recently decided that there is no need to refactor everything to client components but if you still want to merge this, I see no harm in it and I don't believe @amerharb or @richardolsson would object to it either.

**Comment by henrycatalinismith** [comment]
Time to delete this! Was just a proof of concept to grease the wheels of the conversation about architectural directions!

**Pull request closed without merging**